packages/salesforcedx-vscode-core/src/commands/forceRenameLightningComponent.ts

```diff
@@ -0,0 +1,137 @@
+import * as fs from 'fs';
+import * as path from 'path';
+import { format } from 'util';
+import * as vscode from 'vscode';
+import { OUTPUT_CHANNEL } from '../channels';
+import { nls } from '../messages';
+import { FilePathGatherer, SfdxCommandlet, SfdxWorkspaceChecker } from './util';
+
+const RENAME_LIGHTNING_COMPONENT_EXECUTOR = 'force_rename_lightning_component';
+const RENAME_INPUT_PLACEHOLDER = 'rename_component_input_placeholder';
+const RENAME_INPUT_PROMPT = 'rename_component_input_prompt';
+const RENAME_INPUT_DUP_ERROR = 'rename_component_input_dup_error';
+const RENAME_COMPONENT_WARNING = 'rename_component_warning';
+
+export class RenameLwcComponentExecutor extends LibraryCommandletExecutor<string> {
+  private sourceFsPath: string;
+  private responseText: string | undefined;
+  constructor(sourceFsPath: string, responseText: string | undefined) {
+    super(
+      nls.localize(RENAME_LIGHTNING_COMPONENT_EXECUTOR),
+      RENAME_LIGHTNING_COMPONENT_EXECUTOR,
+      OUTPUT_CHANNEL
+    );
+    this.sourceFsPath = sourceFsPath;
+    this.responseText = responseText;
+  }
+
+  public async run(
+    response: ContinueResponse<string>,
+    progress?: vscode.Progress<{ message?: string | undefined; increment?: number | undefined; }>,
+    token?: vscode.CancellationToken
+    ): Promise<boolean> {
+      if (this.sourceFsPath) {
+        if (this.responseText) {
+          renameComponent(this.sourceFsPath, this.responseText);
+          return true;
+        }
+      }
+      return false;
+  }
+}
+
+export async function forceRenameLightningComponent(sourceUri: vscode.Uri) {
+  const sourceFsPath = sourceUri.fsPath;
+  const inputOptions = {
+    placeHolder: nls.localize(RENAME_INPUT_PLACEHOLDER),
+    promopt: nls.localize(RENAME_INPUT_PROMPT)
+  } as vscode.InputBoxOptions;
+
+  const responseText = await vscode.window.showInputBox(inputOptions);
+  if (sourceFsPath) {
+    const commandlet = new SfdxCommandlet(
+      new SfdxWorkspaceChecker(),
+      new FilePathGatherer(sourceUri),
+      new RenameLwcComponentExecutor(sourceFsPath, responseText)
+    );
+    await commandlet.run();
+  }
+}
+
+function renameComponent(sourceFsPath: string, newName: string) {
+  const componentPath = getComponentPath(sourceFsPath);
+  const componentName = path.basename(componentPath);
+  checkForDuplicateName(componentPath, newName);
+  const items = fs.readdirSync(componentPath);
+  if (items) {
+    for (const item of items) {
+      // only rename the file that has same name with component
+      if (isNameMatch(item, componentName, componentPath)) {
+        const newItem = item.replace(componentName, newName);
+        fs.renameSync(
+          path.join(componentPath, item),
+          path.join(componentPath, newItem));
+      }
+    }
+  }
+  const newComponentPath = path.join(path.dirname(componentPath), newName);
+  fs.renameSync(
+    componentPath,
+    newComponentPath);
```

Comment:
Nit pick: when placing args on separate lines (which I like), I think our style guild is to place the closing parenthesis on a new line, like so:
```
fs.renameSync(
    componentPath,
    newComponentPath
);
 ```